test/crux/api_test.clj

```diff
@@ -96,6 +96,17 @@
                 (t/is (= '([:ivan]) result))
                 (t/is (realized? result))))))
 
+        (t/testing "query returning full results"
+          (let [db (.db f/*api*)]
+            (with-open [snapshot (.newSnapshot db)]
+              (let [result (.q db snapshot '{:find [e]
+                                             :where [[e :name "Ivan"]]
+                                             :full-results? true})]
+                (t/is (instance? LazySeq result))
+                (t/is (not (realized? result)))
+                (t/is (= '([{:var e, :value :ivan, :doc {:crux.db/id :ivan, :name "Ivan"}}]) result))
```

Comment:
For consistency with other APIs we probably want name spaced keys here, like `crux.query/var` or similar.